### ISSUE TITLE
All properties of DatabaseClusterAttributes should not be required

### ISSUE THREAD

**Issue by realharry** [aws/aws-cdk#3587]
**Note: for support questions, please first reference our [documentation](https://docs.aws.amazon.com/cdk/api/latest), then use [Stackoverflow](https://stackoverflow.com/questions/ask?tags=aws-cdk)**. This repository's issues are intended for feature requests and bug reports.

* https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseClusterAttributes.html
* https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseCluster.html

* **I'm submitting a ...**
  - [x] :beetle: bug report
  - [ ] :rocket: feature request
  - [ ] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/aws/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*

`DatabaseCluster.from­Database­Cluster­Attributes()` takes the `DatabaseClusterAttributes` interface as a props argument for filtering purposes. But, since every property of the interface is required, the method `from­Database­Cluster­Attributes()` not very useful.

* **What is the expected behavior (or behavior of feature suggested)?**

All properties of DatabaseClusterAttributes should be optional except for a few essential ones, if any.

* **What is the motivation / use case for changing the behavior or adding this feature?**

Currently, there is no way to refer to existing DB from CDK app. `DatabaseCluster.from­Database­Cluster­Attributes()` is the only one I'm aware of, but I cannot use it because all fields are required.

* **Please tell us about your environment:**

  - CDK CLI Version: 1.3.0 (build bba9914)
  - Module Version: 1.3.0
  - OS: OSX Mojave
  - Language: TypeScript

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

**Comment by njlynch** [comment]
Options:

1. **Leave everything as-is:** The `DatabaseClusterAttributes` existing set of properties is necessary to make a fully-functional `IDatabaseCluster` as the interface currently stands, including properly filling in the instance details and `connections` property.

2. **Breaking changes:** Remove most of the properties of `IDatabaseCluster`, only keeping the absolutely critical ones. We could, for example, only make the `clusterIdentifier` required in `DatabaseClusterAttributes` and mandatory in `IDatabaseCluster`, make the `@attributes` optional, and remove the `instanceIdentifiers` and `instanceEndpoints` entirely.

From what I can see, the only usage of the `IDatabaseCluster` interface internal to the CDK is for the Proxy Target:

https://github.com/aws/aws-cdk/blob/377560c99cb48354efe4220aa6ff86fdb45222ff/packages/@aws-cdk/aws-rds/lib/proxy.ts#L423
https://github.com/aws/aws-cdk/blob/377560c99cb48354efe4220aa6ff86fdb45222ff/packages/@aws-cdk/aws-rds/lib/proxy.ts#L58-L75

The latter won't even work with an imported cluster if I'm reading it correctly.

So there are no internal references preventing us from breaking the interface, but any and all customers who already use and rely on this interface would be impacted. 

Tagging @skinny85  and @shivlaks  -- we'll discuss internally and make a decision on how to proceed.

**Comment by mmassenzio** [comment]
I see a PR was merged and this issue closed - however, I still see this behavior with CDK 1.134.0 (when trying to use in Java, `DatabaseCluster.fromDatabaseClusterAttributes()` with any of the properties unset, a `NPE` is thrown).

Which version of the CDK will the fix work on?
Also, it would be great if this method usage (or the `DBClusterAttributes` class) could be documented better.

Thanks.

**Comment by skinny85** [comment]
@mmassenzio yes, this should definitely work in CDK `1.34.0`.

Can you show your code, and the exact error message that you get?

**Comment by kundkingan** [comment]
Same here, not optional in `1.34.0`

**Comment by skinny85** [comment]
@kundkingan same ask as above 😉.

**Comment by kundkingan** [comment]
@skinny85 I checked my .lock file and it is 1.134.0 and in docs those properties are not optional. I guess the change have not been released?

**Comment by massenz** [comment]
sorry, folks - have been busy on other stuff this week and the repo I'm working on is private, can't share.
Please let me get something together that I can share in a gist.

**Comment by skinny85** [comment]
The change definitely has been released - for a long time, actually (since `v1.63.0`, September 2020).

@kundkingan can you show me your code, and the exact error that you get?

**Comment by tomjowitt** [comment]
I am still getting this error with the Go CDK:

```bash
cdk version
2.10.0 (build e5b301f)
```

```go
	db := awsrds.DatabaseInstance_FromDatabaseInstanceAttributes(
		stack,
		jsii.String("xxxxRDSLookup"),
		&awsrds.DatabaseInstanceAttributes{
			InstanceIdentifier:      jsii.String("xxxxxxxx"),
		},
	)
```

Which results in the error:

```
panic: "Missing required properties for monocdk.aws_rds.DatabaseInstanceAttributes: instanceEndpointAddress, port, securityGroups"
```

I also tried adding nil values which didn't fix it.

```go
	db := awsrds.DatabaseInstance_FromDatabaseInstanceAttributes(
		stack,
		jsii.String("xxxxRDSLookup"),
		&awsrds.DatabaseInstanceAttributes{
			InstanceEndpointAddress: nil,
			InstanceIdentifier:      jsii.String("xxxxxxxx"),
			Port:                    nil,
			SecurityGroups:          nil,
		},
	)
```
